application/src/main/java/org/togetherjava/tjbot/features/javamail/RSSHandlerRoutine.java

```diff
@@ -0,0 +1,311 @@
+        }
+        return true;
+    }
+
+    /**
+     * Attempts to find a text channel from a given RSS feed configuration.
+     */
+    private Optional<TextChannel> getTextChannelFromFeed(JDA jda, RSSFeed feed) {
+        // Attempt to find the target channel
+        if (feed.targetChannelPattern() != null) {
+            return jda.getTextChannelCache()
+                .stream()
+                .filter(channel -> targetChannelPatterns.get(feed).test(channel.getName()))
+                .findFirst();
+        }
+
+        // If the target channel was not found, use the fallback
+        return jda.getTextChannelCache()
+            .stream()
+            .filter(channel -> defaultChannelPattern.test(channel.getName()))
+            .findFirst();
+    }
+
+    /**
+     * Provides the {@link EmbedBuilder} from an RSS item used for sending RSS posts.
+     */
+    private static EmbedBuilder constructEmbedMessage(Item item, RSSFeed feedConfig) {
+        final EmbedBuilder embedBuilder = new EmbedBuilder();
+        String title = item.getTitle().orElse("No title");
+        String titleLink = item.getLink().orElse("");
+        Optional<String> rawDescription = item.getDescription();
+
+        // Set the item's timestamp to the embed if found
+        item.getPubDate()
+            .ifPresent(date -> embedBuilder
+                .setTimestamp(getLocalDateTime(date, feedConfig.dateFormatterPattern())));
+
+        embedBuilder.setTitle(title, titleLink);
+        embedBuilder.setAuthor(item.getChannel().getLink());
+
+        // Process embed's description if a raw description was provided
+        if (rawDescription.isPresent() && !rawDescription.get().isEmpty()) {
+            Document fullDescription =
+                    Jsoup.parse(StringEscapeUtils.unescapeHtml4(rawDescription.get()));
+            StringBuilder finalDescription = new StringBuilder();
+            fullDescription.body()
+                .select("*")
+                .forEach(p -> finalDescription.append(p.text()).append(". "));
+
+            embedBuilder
+                .setDescription(StringUtils.abbreviate(finalDescription.toString(), MAX_CONTENTS));
+            return embedBuilder;
+        }
+
+        // Fill the description with a placeholder if the description was empty
+        embedBuilder.setDescription("No description");
+        return embedBuilder;
+    }
+
+    /**
+     * Fetches a list of {@link Item} from a given RSS url.
+     */
+    private static List<Item> fetchRss(String rssUrl) {
+        try {
+            return RSS_READER.read(rssUrl).toList();
+        } catch (Exception e) {
+            logger.warn("Could not fetch RSS from URL ({})", rssUrl);
+            return List.of();
+        }
+    }
+
+    /**
+     * Helper function for parsing a given date value to a {@link ZonedDateTime} with a given
+     * format.
+     */
+    private static ZonedDateTime getLocalDateTime(@Nullable String date, @NotNull String format) {
```

Comment:
The name is very misleading, because LocalDateTime is an existing class in the same package as ZonedDateTime.
So your method is saying that it will result in a LocalDateTime but returns a ZonedDateTime ?